optimum/onnxruntime/modeling_decoder.py

```diff
@@ -596,50 +639,34 @@ def _from_pretrained(
         else:
             init_cls = ORTModelForCausalLM
 
-        if generation_config is None:
-            try:
-                generation_config = GenerationConfig.from_pretrained(
-                    model_id,
-                    cache_dir=cache_dir,
-                    force_download=force_download,
-                    local_files_only=local_files_only,
-                    token=token,
-                    revision=revision,
-                    subfolder=subfolder,
-                )
-            except OSError:
-                logger.info(
-                    "Generation config file not found, using a generation config created from the model config."
-                )
-
         return init_cls(
-            model=model,
             config=config,
+            session=session,
             use_io_binding=use_io_binding,
+            generation_config=generation_config,
             model_save_dir=model_save_dir,
             use_cache=use_cache,
-            generation_config=generation_config,
         )
 
     @classmethod
-    def _from_transformers(
+    def _export(
         cls,
-        model_id: str,
+        model_id: Union[str, Path],
         config: "PretrainedConfig",
+        # export options
+        task: Optional[str] = None,
```

Comment:
```suggestion
```
what do you think about deprecating `task` completely ? (raisin an error if provided)